TumourModelParticleFiltering/TumourModel_ParticleFiltering.m

```diff
@@ -102,37 +104,40 @@
                 res(j, i) = (PFTumourVolume(i)- data);
             end
      end
-     
+
     % records vector of residuals for the paticles and stores them in a
     % matrix for the different data time points
     res_mat(t+1, :, :) = res;
     tvol_mat(t+1, :)   = PFTumourVolume;
     tvol_mean_predicted(t) = mean(PFTumourVolume)
     t
-    
+
     for i = 1:n
         log_weight(t, i) = 0;
         for j = 1:n_mice
-            log_weight(t, i) = log_weight(t, i) - 1/(4*2500) * res(j, i).^2;
+            log_weight(t, i) = log_weight(t, i) - 1/(2 * obsSigma^2) * res(j, i).^2;
         end
-        non_norm_weight(t, i) = exp(log_weight(t, i));
+
     end
+    % To avoid underflow when we move from log space
+    max_log_weight = max(log_weight(t, :));
+    non_norm_weight(t, :) = exp(log_weight(t, :) - max_log_weight);
```

Comment:
Suppose the log weight was -800 but exp(-800) is 0.0. So if all the particles had log weights of -800 we'd get a NaN. In this case the max would also be -800 so by subtracting we get a weight of 0.0 for each particle and exp (0.0) = e. Normalising means this trick doesn't change the mathematical answer but it does avoid underflow.